File: 151001/Novik/RESTLab1/REST/Mapper/UserMapper.cs
using AutoMapper;
using REST.Models.DTOs.RequestIn;
using REST.Models.DTOs.Responses;
using REST.Models.Entity;

namespace REST.Mapper;

public class UserMapper : Profile
{
    public UserMapper()
    {
        CreateMap<User, UserResponseTo>();
        CreateMap<UserRequestTo, User>();
    }
    
}

File: 151001/Trubchik/dc/publisher/src/main/java/by/bsuir/ilya/dto/UserMapper.java
package by.bsuir.ilya.dto;

import by.bsuir.ilya.Entity.User;
import org.mapstruct.Mapper;
import org.mapstruct.factory.Mappers;

@Mapper
public interface UserMapper {
    UserMapper INSTANCE = Mappers.getMapper( UserMapper.class );



    UserResponseTo userToUserResponseTo(User user);

    UserRequestTo userToUserRequestTo(User user);

    User userResponseToToUser(UserResponseTo userResponseTo);

    User userRequestToToUser(UserRequestTo userRequestTo);
}
